src/base.iced

```diff
@@ -173,6 +178,16 @@ class Base
 
   #------
 
+  _v_include_full_pgp_hash : -> false
+  _v_pgp_km_to_hash : () -> @km()
+  _v_pgp_hash_dest : (body) -> body.key
+
+  full_pgp_sha256: () ->
+    if '_cached_pgp_sha256' of @ then @_cached_pgp_sha256
+    else (key = @_v_pgp_km_to_hash()?.armored_pgp_public)? and (@_cached_pgp_sha256 = hash_pgp_key key) or null
```

Comment:
I cache the hash because this function is public and could be called again externally (`key/add.json` calls it). There's a theoretical risk that someone could swap out the underlying `KeyManager` and the cache won't be invalidated, but I think that's okay to ignore for the moment.